ocaml/testsuite/tests/typing-layouts-arrays/test_int32_u_array.ml

```diff
@@ -46,6 +46,13 @@ module Int32_u_array0 : Gen_u_array.S0
   type ('a : any) array_t = 'a array
   type element_arg = unit -> element_t
   type t = element_t array
+
+  let max_length =
+    match Sys.backend_type with
+    | Bytecode -> Sys.max_array_length
+    | Native -> Sys.max_unboxed_int32_array_length
+    | Other _ -> assert false
```

Comment:
Not sure about this - maybe this conditional should be directly in `Sys`?

Reply:
Sure, moved the conditionals to sys